src/osd/OSD.cc

```diff
@@ -8184,7 +8223,9 @@ void OSD::ShardedOpWQ::_process(uint32_t thread_index, heartbeat_handle_d *hb )
       return;
     }
   }
+  sdata->ql_accum += sdata->pqueue.length(); 
```

Comment:
Sadly PrioritizedQueue::length isn't a O(1) implementation. We need to consider to avoid to get length each time or modify PrioritizedQueue::length